src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSearchValues.cs

```diff
@@ -0,0 +1,314 @@
+            if (argument is IArrayCreationOperation arrayCreation)
+            {
+                // text.IndexOfAny(new[] { 'a', 'b', 'c' })
+                return IsConstantByteOrCharSZArrayCreation(arrayCreation, out _);
+            }
+            else if (argument is IFieldReferenceOperation fieldReference)
+            {
+                // readonly char[] Values = new char[] { 'a', 'b', 'c' };
+                // readonly char[] Values = "abc".ToCharArray();
+                // readonly char[] Values = ConstString.ToCharArray();
+                // text.IndexOfAny(Values)
+                return
+                    IsConstantByteOrCharSZArrayFieldReference(fieldReference, out int length) &&
+                    length >= MinLengthWorthReplacing;
+            }
+            else if (argument is IInvocationOperation invocation)
+            {
+                // text.IndexOfAny("abc".ToCharArray())
+                // text.IndexOfAny(StringConst.ToCharArray())
+                return IsConstantStringToCharArrayInvocation(invocation, out _);
+            }
+
+            return false;
+        }
+
+        private bool AreConstantValuesWorthReplacing(IOperation argument, INamedTypeSymbol readOnlySpanType)
+        {
+            if (argument is IConversionOperation conversion)
+            {
+                if (IsConstantStringLiteralOrReference(conversion.Operand, out string? value))
+                {
+                    // text.IndexOfAny("abc")
+                    // or
+                    // const string ValuesLocalOrField = "abc";
+                    // text.IndexOfAny(ValuesLocalOrField)
+                    return value.Length >= MinLengthWorthReplacing;
+                }
+                else if (conversion.Operand is IArrayCreationOperation arrayCreation)
+                {
+                    // text.IndexOfAny(new[] { 'a', 'b', 'c' })
+                    return
+                        IsConstantByteOrCharSZArrayCreation(arrayCreation, out int length) &&
+                        length >= MinLengthWorthReplacing;
+                }
+                else if (conversion.Operand is IFieldReferenceOperation fieldReference)
+                {
+                    // readonly char[] Values = new char[] { 'a', 'b', 'c' };
+                    // readonly char[] Values = "abc".ToCharArray();
+                    // readonly char[] Values = ConstString.ToCharArray();
+                    // text.IndexOfAny(Values)
+                    return
+                        IsConstantByteOrCharSZArrayFieldReference(fieldReference, out int length) &&
+                        length >= MinLengthWorthReplacing;
+                }
+                else if (conversion.Operand is IInvocationOperation invocation)
+                {
+                    // text.IndexOfAny("abc".ToCharArray())
+                    // text.IndexOfAny(StringConst.ToCharArray())
+                    return IsConstantStringToCharArrayInvocation(invocation, out _);
```

Comment:
All of this code seems to be identical to the code within `AreConstantValuesWorthReplacingForStringIndexOfAny`. Consider sharing code in a helper